Api/Services/Markups/MarkupBonusMaterializationService.cs

```
@@ -0,0 +1,126 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace HappyTravel.Edo.Api.Services.Markups
+{
+    public class MarkupBonusMaterializationService  : IMarkupBonusMaterializationService
+    {
+        public MarkupBonusMaterializationService(EdoContext context, IDateTimeProvider dateTimeProvider)
+        {
+            _context = context;
+            _dateTimeProvider = dateTimeProvider;
+        }
+
+
+        public Task<List<int>> GetForMaterialize(DateTime dateTime)
+        {
+            var query =
+                from appliedMarkup in _context.AppliedBookingMarkups
+                join booking in _context.Bookings on appliedMarkup.ReferenceCode equals booking.ReferenceCode
+                join policy in _context.MarkupPolicies on appliedMarkup.PolicyId equals policy.Id
+                where 
+                    booking.Status == BookingStatuses.Confirmed &&
+                    (booking.PaymentStatus == BookingPaymentStatuses.Authorized ||
+                    booking.PaymentStatus == BookingPaymentStatuses.Captured) &&
+                    booking.CheckOutDate.Date >= dateTime && 
+                    appliedMarkup.Paid == null && 
+                    policy.ScopeType == MarkupPolicyScopeType.Agent
+                select appliedMarkup.Id;
+
+            return query.ToListAsync();
+        }
+
+
+        public async Task<Result<BatchOperationResult>> Materialize(List<int> markupsForMaterialization)
+        {
+            foreach (var materializationData in await GetData(markupsForMaterialization))
+            {
+                await Result.Success(materializationData)
+                    .BindWithTransaction(_context, m => Result.Success(m)
+                        .Map(ApplyBonus)
+                        .Map(MarkAsPaid)
+                        .Tap(WriteLog)
+                    );
+            }
+            
+            return new BatchOperationResult($"{markupsForMaterialization.Count} markups materialized", false);
```

Errors are swallowed, this  is not nice. It there are any failure results, hasErrors should be set to `true`

---

No Result in flow